src/Altinn.App.Core/Implementation/AppResourcesSI.cs

```diff
@@ -104,67 +109,52 @@ public byte[] GetText(string org, string app, string textResource)
         /// <inheritdoc />
         public Application GetApplication()
         {
-            // Cache application metadata
-            if (_application != null)
-            {
-                return _application;
-            }
-
-            string filedata = string.Empty;
-            string filename = _settings.AppBasePath + _settings.ConfigurationFolder + _settings.ApplicationMetadataFileName;
             try
             {
-                if (File.Exists(filename))
-                {
-                    filedata = File.ReadAllText(filename, Encoding.UTF8);
-                }
-
-                _application = JsonConvert.DeserializeObject<Application>(filedata)!;
-                return _application;
+                var task = Task.Run(async () => await _appMetadata.GetApplicationMetadata());
+                task.Wait();
+                return task.Result;
             }
-            catch (Exception ex)
+            catch (AggregateException ex)
             {
-                _logger.LogError("Something went wrong when fetching application metadata. {0}", ex);
-                return null;
+                throw new ApplicationConfigException("Failed to read application metadata", ex.InnerException ?? ex);
             }
         }
 
         /// <inheritdoc/>
         public string? GetApplicationXACMLPolicy()
         {
-            string filename = _settings.AppBasePath + _settings.ConfigurationFolder + _settings.AuthorizationFolder + _settings.ApplicationXACMLPolicyFileName;
-            try
+            var task = Task.Run<string?>(async () => await _appMetadata.GetApplicationXACMLPolicy());
+            task.Wait();
+            if (task.IsCompletedSuccessfully)
             {
-                if (File.Exists(filename))
-                {
-                    return File.ReadAllText(filename, Encoding.UTF8);
-                }
-            }
-            catch (Exception ex)
-            {
-                _logger.LogError("Something went wrong when fetching XACML Policy. {0}", ex);
+                return task.Result;
             }
 
```

Comment:
Same here, worth a comment on why we allow swallowing this particular exception while the others are thrown.